### ISSUE TITLE
INCR can reset your timeout value

### ISSUE THREAD

**Issue by jpulec** [niwinz/django-redis#181]
There is a race condition in the implementation of `DefaultClient._incr()` and it can result in timeout values being unset for the specified key. The redis docs state that if INCR is called when there is no existing key, the key will be created and have a value of 0. It will also be created as a nonvolatile key (ttl of -1).

```python
try:
    if not client.exists(key):
        raise ValueError("Key '%s' not found" % key)

    try:
        value = client.incr(key, delta)
```

In the above snippet from `default.py`, if the key exists when the first if statement is executed a ValueError will not be raised. However if between the if statement being executed and the `client.incr(key, delta)` statement running the key is deleted or expires, the call to INCR will create the key with a ttl = -1, which is not the expected behavior since calling `cache.incr()` from django should never affect the timeout.

Happy to provide some more info, if it will help.

**Comment by niwinz** [comment]
Thanks for report that.

I understand the problem, maybe it can be solved with "transactions".

**Comment by niwinz** [comment]
I don't have time for test this properly and add the correct fix. If some one have time for it, the PR is very welcome. I think it can be implemented just unsing `WATCH` + `EXEC`.

**Comment by prokaktus** [comment]
@niwinz 
Unfortunately, it will not work, because of
> Note that if you WATCH a volatile key and Redis expires the key after you WATCHed it, EXEC will still work.
http://redis.io/topics/transactions

**Comment by niwinz** [comment]
Fixed in master and the 4.4.4
